Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -306,9 +307,8 @@ SkPaint GraphicsContextSkia::createFillPaint(std::optional<Color> fillColor) con
         paint.setColor(SkColorSetARGB(a, r, g, b));
     }
 
-    // Outset shadow
-    // FIXME: Don't add the effect if the shadow is inset
-    if (hasDropShadow()) {
+    // Drop shadow
+    if (hasDropShadow() && shadowStyle != ShadowStyle::None) {
```

Comment:
I'm no longer sure it's a good idea to handle this here. We know what operations can have a shadow and what kind of shadow to use in every case, so maybe it would be better to add helper functions to create the SkImageFilter and add it to the SkPaint in the operations after calling createFillPaint().

Reply:
Moved it to a helper method